### ISSUE TITLE
Overflow error in EuclideanRing

### ISSUE THREAD

**Issue by jacereda** [purescript-contrib/purescript-quickcheck-laws#29]
```
Checking 'Submultiplicative euclidean function' law for EuclideanRing
999/1000 test(s) passed.
/Users/jacereda/src/purescript/purescript-quickcheck-laws/output/Control.Monad.Eff.Exception/foreign.js:29
    throw e;
    ^

Error: Test 90 (seed 2114546612) failed: 
Test returned false
```

**Comment by hdgarrood** [comment]
Which type is this? Also, how do you know it's an overflow error?

**Comment by jacereda** [comment]
`Int` and I assume it's an overflow because the product can easily overflow with two arbitrary ints.

**Comment by jacereda** [comment]
To reproduce it easily, just increase the number of iterations for `submultiplicative` in `checkEuclideanRing`, something like 10000 suffice.

**Comment by hdgarrood** [comment]
Right, I guess we should just remove that test from this repo's test suite then.

**Comment by jacereda** [comment]
Or perform an overflow check?

**Comment by hdgarrood** [comment]
We can't do that with just a `EuclideanRing` constraint, though, surely?

**Comment by jacereda** [comment]
nope, I guess `Ord` and `Bounded` would be required.

**Comment by hdgarrood** [comment]
Right - and there are perfectly good euclidean rings which are neither `Ord` nor `Bounded`; gaussian integers and polynomials over a field, for example.

**Comment by jacereda** [comment]
Looks like we can do without `Bounded`:

https://github.com/garyb/purescript-quickcheck-laws/pull/30

**Comment by JordanMartinez** [comment]
As discussed in https://github.com/purescript-contrib/purescript-quickcheck-laws/pull/30#issuecomment-923048466
> Looks like the next action is to drop the overflow check and instead wrap Int in a newtype that limits how large the initial generated Int can be, so that an overflow error cannot occur. That would fix the false positive.
